ArduPlane/GCS_Mavlink.cpp

```diff
@@ -794,13 +794,169 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
+        }
+         // the requested alt data might be relative or absolute
+        float new_target_alt = packet.z * 100;
+        float new_target_alt_rel = packet.z * 100 + plane.home.alt;
+
+         // only global/relative/terrain frames are supported
+        switch(packet.frame) {
+            case MAV_FRAME_GLOBAL_RELATIVE_ALT: {
+                if   (fabsf(plane.guided_state.target_alt - new_target_alt_rel)<0.001f ) { // compare two floats as near-enough
+                    // reject duplicate altitude change requests
+                    return MAV_RESULT_TEMPORARILY_REJECTED;
+                }
+                plane.guided_state.target_alt = packet.z * 100 + plane.home.alt;
+                break;
+            }
+            case MAV_FRAME_GLOBAL: {
+                if   (fabsf(plane.guided_state.target_alt - new_target_alt)<0.001f ) {  // compare two floats as near-enough
+                    // reject duplicate altitude change requests
+                    return MAV_RESULT_TEMPORARILY_REJECTED;
+                }
+                plane.guided_state.target_alt = packet.z * 100;
+                break;
+            }
+            default:
+                // this wasn't a mission_item, so no forms of frame nacks are supported, MAV_RESULT_UNSUPPORED is the best we can do
+                return MAV_RESULT_UNSUPPORTED;
+        }
+
+        plane.guided_state.target_alt_frame = packet.frame;
+        plane.guided_state.last_target_alt = plane.current_loc.alt; // FIXME: Reference frame is not corrected for here
+        plane.guided_state.target_alt_time_ms = AP_HAL::millis();
+
+        if (is_zero(packet.param3)) {
+            // the user wanted /maximum acceleration, pick a large value as close enough
+            plane.guided_state.target_alt_accel = 1000.0;
+        } else {
+            plane.guided_state.target_alt_accel = fabsf(packet.param3);
+        }
+
+         // assign an acceleration direction
+        if (plane.guided_state.target_alt < plane.current_loc.alt) {
+            plane.guided_state.target_alt_accel *= -1.0f;
+        }
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_HEADING: {
+
+        // command is only valid in guided mode
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // don't accept packets outside of [0-360] degree range
+        if (packet.param2 < 0.0f || packet.param2 >= 360.0f) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        float new_target_heading = radians(wrap_180(packet.param2));
+
+        // reject duplicate heading requests if our target-heading has been set to anything except it's default value of -1
+        if (( plane.guided_state.target_heading > -0.001f ) && (fabsf(new_target_heading - plane.guided_state.target_heading)<0.001f)) { // compare two floats as near-enough
+            return MAV_RESULT_TEMPORARILY_REJECTED;
```

Comment:
why reject? same Q as above

Reply:
same answer as above, we don't want to handle the 'repeated' packet any further, and we don't want to reset the internal deltas() that have already started to accumulate.